code/game/objects/structures/fusioncell.dm

```diff
@@ -0,0 +1,83 @@
+/obj/structure/fusioncell
+	name = "Fusion Cell"
+	desc = "A simple way to store radioactive energy with many capabilities."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "oldshieldoff"
+	density = FALSE
+	anchored = FALSE
+	max_integrity = INFINITY
+	var/emitting = FALSE
+	var/radiation_count = 0
+	var/emittingmount = 0
+	var/i
+	var/datum/looping_sound/geiger/soundloop
+
+
+	/obj/structure/fusioncell/attackby(obj/item/I, mob/user, params)
+		if(I.tool_behaviour == TOOL_WRENCH)
+			default_unfasten_wrench(user, I, time = 5)
+		return
+
+	/obj/structure/fusioncell/proc/update_sound()
+		var/datum/looping_sound/geiger/loop = soundloop
+		if(radiation_count < 3500 || !emitting)
+			loop.stop()
+			return
+		loop.start()
+
+	/obj/structure/fusioncell/interact(mob/user)
+		if(emitting == FALSE)
+			to_chat(user, "You turn the cell on")
+			rad_insulation = RAD_FULL_INSULATION //prevents infinite rads ie cant emit rads and gain rads at the same time
```

Comment:
Do you know that radiation insulation happens after rad_act? Even if you intended the radiation battery to be an extremely powerful radiation sponge that consumes all radiation, but the charging will happen to *every* cell on that turf. You should make this structure `dense = TRUE`, or devise other solutions.